src/transformers/models/mllama/modeling_mllama.py

```diff
@@ -1618,6 +1741,9 @@ def forward(
 
         hidden_states = inputs_embeds
 
+        if use_cache and past_key_values is None:
+            past_key_values = DynamicCache()
+
```

Comment:
this does make sense as it's helping users, and an old APi, but let's promote init of a cache and passing it! 🤗

Reply:
We could but still I think it is a lot easier if users want a forward pass with cache, and do not want extra lines of code for importing and passing the cache object. So i think we'd better keep the default cache for now